sdk-actors/src/main/java/io/dapr/actors/runtime/DaprStateAsyncProvider.java

```diff
@@ -177,22 +180,17 @@
    * returned as-is without deserializing first.
    *
    * @param raw Bytes received from Dapr.
-   * @return Corrected byte[].
+   * @param clazz Class to be deserialized to.
+   * @return Deserialized object.
+   * @throws IOException If could not deserialize.
    */
-  private byte[] fixDaprStateResponse(byte[] raw) {
-    if (raw == null) {
-      return raw;
-    }
-
-    if (raw.length == 0) {
-      return raw;
-    }
-
+  private <T> T deserializeDaprStateResponse(byte[] raw, Class<T> clazz) throws IOException {
     try {
-      return OBJECT_MAPPER.readValue(raw, String.class).getBytes();
+      // Try to deserialize after quoted string first.
+      return this.stateSerializer.deserialize(OBJECT_MAPPER.readValue(raw, String.class).getBytes(), clazz);

```

Comment:
How does this work?  Can you provide an example of what format it would have when this works and another for when it uses the other case?

Reply:
I just realized we don't need this anymore. This was needed because the server side was running with an older version. Actually, the data is much cleaner now.

Reply:
![image](https://user-images.githubusercontent.com/130954/74078253-dc8bbe80-49dc-11ea-8a21-aae31936178c.png)